src/services/dbService.ts

```diff
@@ -3,20 +3,43 @@ import { injectable, inject } from 'inversify'
 import * as Knex from 'knex'
 import RDSService from './aws/rdsService'
 import TYPES from '../types'
+import * as postgis from 'knex-postgis'
 
 @injectable()
 export default class DBService {
+  private connection: Knex
+  private knexPostgis: postgis.KnexPostgis
+
   public constructor(
     @inject(TYPES.RDSService) private rds: RDSService,
     @inject(TYPES.Knex) private knexConfig: Knex.Config) {
   }
 
-  public async connect(): Promise<Knex> {
-    const config: Knex.Config = this.knexConfig
+  public postgis(): postgis.KnexPostgis {
+    if (!this.knexPostgis) {
+      this.knexPostgis = postgis(this.connection)
+    }
+    return this.knexPostgis
+  }
 
-    await this.configureConnection(<Knex.PgConnectionConfig>this.knexConfig.connection)
+  public async knex(): Promise<Knex> {
+    if (!this.connection) {
+      await this.connect()
+    }
 
-    return Knex(config)
+    return this.connection
+  }
+
+  public async destroy(): Promise<void> {
+    if (!this.connection) {
```

Comment:
I think this should be 
```suggestion
    if (this.connection) {
```